code/modules/xenoarcheaology/effects/vampire.dm

```diff
@@ -0,0 +1,86 @@
+
+/datum/artifact_effect/vampire
+	name = "vampire"
+	effect_type = EFFECT_ORGANIC
+	var/last_bloodcall = 0
+	var/bloodcall_interval = 50
+	var/last_eat = 0
+	var/eat_interval = 100
+	var/charges = 0
+	var/list/nearby_mobs = list()
+
+/datum/artifact_effect/vampire/proc/bloodcall(var/mob/living/carbon/human/M)
+	last_bloodcall = world.time
+	if(istype(M))
+		playsound(holder.loc, pick('sound/hallucinations/wail.ogg','sound/hallucinations/veryfar_noise.ogg','sound/hallucinations/far_noise.ogg'), 50, 1, -3)
+
+		var/target = pick(M.organs_by_name)
+		M.apply_damage(rand(5, 10), SEARING, target)
+		to_chat(M, "<span class='critical'>The skin on your [parse_zone(target)] feels like it's ripping apart, and a stream of blood flies out.</span>")
+		var/obj/effect/decal/cleanable/blood/splatter/animated/B = new(M.loc)
+		B.basecolor = M.species.get_blood_colour(M)
+		B.color = M.species.get_blood_colour(M)
+		B.target_turf = pick(range(1, get_turf(holder)))
+		B.blood_DNA = list()
+		B.blood_DNA[M.dna.unique_enzymes] = M.dna.b_type
+		M.vessel.remove_reagent("blood",rand(25,50))
+
+/datum/artifact_effect/vampire/DoEffectTouch(var/mob/user)
+	bloodcall(user)
+	DoEffectAura()
+
+/datum/artifact_effect/vampire/DoEffectAura()
+	nearby_mobs.Cut()
+
+	var/turf/T = get_turf(holder)
+
+	for(var/mob/living/L in oview(effectrange, T))
```

Comment:
immune if you manage to make it on top of the artifact, then

Reply:
Artifact batteries.